utils/s3/client.cc

```diff
@@ -816,11 +816,18 @@ data_sink client::make_upload_jumbo_sink(sstring object_name, std::optional<unsi
 class client::readable_file : public file_impl {
     shared_ptr<client> _client;
     sstring _object_name;
+    std::optional<stats> _stats;
 
     [[noreturn]] void unsupported() {
         throw_with_backtrace<std::logic_error>("unsupported operation on s3 readable file");
     }
 
+    future<> maybe_update_stats() {
+        if (!_stats) {
+            _stats = co_await _client->get_object_stats(_object_name);
+        }
```

Comment:
One of those cases where a coroutine is a pessimization.

Reply:
Yes. Made it future-returning in v2